Support passing URLs as cmdline arguments

bionade24 opened the pull request:
I've not renamed the vars with file in it in all files as I want your opinion on that 1st. I settled for  `mediaUrl` and `mediaRessource`, to make it more obvious that it may also be a stream and "medium" sounded more too much like esoteric magic.

Spaces in URLs don't work yet.

bionade24 committed:
Support URLs as command line argument.

This makes it possible to use smtube with kokovp instead of smplayer.

bionade24 commented:
Force-pushed to drop commits from other PR I accidentally included.

bionade24 committed:
Formatting

bionade24 committed:
Improvement, to be squashed.

bionade24 commented:
@brainrom Could I please get your opinion on this?

brainrom commented on playercontroller.cpp:
```diff
@@ -188,7 +188,7 @@ void PlayerController::handleFileLoad()
 
         if (t.isExternal)
-            t.filename = p->getProp(trackAddr + "external-filename").toString();
+            t.mediaUrl = p->getProp(trackAddr + "external-mediaUrl").toString();
```
It wont work. `external-filename` is the mpv's keyword.

brainrom commented on program_arg.h:
program_arg is over-engineering. SingleInstance is designed to be as simple as possible stream communication and doesn't require additional structure.

brainrom commented on playercontroller.h:
Massive rename without exact purpose. file can be accessed by url as well.

brainrom commented on main.cpp:
```diff
@@ -36,6 +37,7 @@ int main(int argc, char *argv[])
     parser.addHelpOption();
     parser.addVersionOption();
     parser.addPositionalArgument("media", QCoreApplication::translate("KokoVP", "Media files to play"));
+    parser.addPositionalArgument("urls", QCoreApplication::translate("KokoVP", "URLs to play"));
```
How multiple position arguments lists should work at all? Maybe just accept media by local file paths and by urls?

bionade24 replied:
May you please elaborate on this? I don't understand what you mean by this.

brainrom commented on helper.cpp:
Looks really useful for achieving accepting URLs. Needs testing and fixing mentioned issue, but anyway.

brainrom reviewed (commented):
A lot of meaningless and confusing variable/methods renames, one mpv keyword is broken.
program_arg is over-engineering, which greatly limits class usage scope.
New URL converter is really needed and maybe will allow to accept URLs without any other changes.

bionade24 committed:
Add missing header in singleinstance.cpp

bionade24 committed:
Add Open URL button to Open menu.

bionade24 committed:
Conditionally start playback of opened media in existing instance.

If something is already currently playing on the existing instance, the
media gets added to the playlist, like before. If nothing is playing,
start playback of the added media.

bionade24 committed:
Revert renaming from file to media(Ressource)

To be squashed.

bionade24 committed:
Add missing QMessageBox header

bionade24 committed:
More cleanup

To be squashed.

bionade24 committed:
Cleanup, to be squashed
